### PR TITLE
Unit tests for Path.Join overloads

### PR DESCRIPTION
**Pull request by jbhensley**
Unit tests for Path.Join overloads in corclr [PR 24307](https://github.com/dotnet/coreclr/pull/24307). corefx issue #32312

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs [hunk 1]
```diff
@@ -117,5 +117,77 @@ public void TryJoinThreePaths(string path1, string path2, string path3, string e
...
+            { "a", $"{AltSep}b", "", "", $"a{AltSep}b" },
+            { "a", "", $"{AltSep}b", "", $"a{AltSep}b" },
+            { "a", "", "", $"{AltSep}b", $"a{AltSep}b" },
+            { null, null, null, null, "" },
+            { "a", null, null, null, "a" },
+            { null, "a", null, null, "a" },
+            { null, null, "a", null, "a" },
+            { null, null, null, "a", "a" },
+            { "a", null, "b", null, $"a{Sep}b" },
+            { "a", null, null, "b", $"a{Sep}b" }
```
Comment (Anipik) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: can you add some test cases where the strings are starting with a sep ?

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs [hunk 2]
```diff
@@ -117,5 +117,77 @@ public void TryJoinThreePaths(string path1, string path2, string path3, string e
...
+        [Theory, MemberData(nameof(TestData_JoinThreePaths))]
+        public void JoinStringArray_3(string path1, string path2, string path3, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2, path3 }));
+        }
+
+        [Theory, MemberData(nameof(TestData_JoinFourPaths))]
+        public void JoinStringArray_4(string path1, string path2, string path3, string path4, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2, path3, path4 }));
```
Comment (Anipik) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: can we have test for more than 4 as well ? maybe just pass the all the paths twice

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs [hunk 3]
```diff
@@ -117,5 +117,77 @@ public void TryJoinThreePaths(string path1, string path2, string path3, string e
...
+        [Theory, MemberData(nameof(TestData_JoinTwoPaths))]
+        public void JoinStringArray_2(string path1, string path2, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2 }));
+        }
+
+        [Theory, MemberData(nameof(TestData_JoinThreePaths))]
+        public void JoinStringArray_3(string path1, string path2, string path3, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2, path3 }));
```
Comment (Anipik) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: can you please add test when the param array itself is null ?

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs [hunk 4]
```diff
@@ -189,5 +205,12 @@ public void JoinStringArray_4(string path1, string path2, string path3, string p
         {
             Assert.Equal(expected, Path.Join(new string[] { path1, path2, path3, path4 }));
         }
+
+        [Theory, MemberData(nameof(TestData_JoinFourPaths))]
+        public void JoinStringArray_8(string path1, string path2, string path3, string path4, string _)
+        {
+            string fourJoined = Path.Join(path1, path2, path3, path4);
```
Comment (Anipik) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: you can just use the ```expected``` string here

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs [hunk 5]
```diff
@@ -8,6 +8,15 @@ namespace System.IO.Tests
 {
     public class PathTests_Join : PathTestsBase
     {
+        public static TheoryData<string, string> TestData_JoinOnePath = new TheoryData<string, string>
+        {
+            { "", "" },
+            { Sep, Sep },
+            { AltSep, AltSep },
+            { "a", "a" },
+            { null, ""}
```
Comment (jakub-klima) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]:
```suggestion
            { null, "" }
```

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs [hunk 6]
```diff
@@ -117,5 +126,111 @@ public void TryJoinThreePaths(string path1, string path2, string path3, string e
...
+        [Theory, MemberData(nameof(TestData_JoinFourPaths))]
+        public void JoinStringArray_4(string path1, string path2, string path3, string path4, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2, path3, path4 }));
+        }
+
+        [Theory, MemberData(nameof(TestData_JoinFourPaths))]
+        public void JoinStringArray_8(string path1, string path2, string path3, string path4, string fourJoined)
+        {   
+            Assert.Equal(Path.Join(fourJoined, fourJoined) , Path.Join(new string[] { path1, path2, path3, path4, path1, path2, path3, path4 }));
```
Comment (jakub-klima) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]:
```suggestion
            Assert.Equal(Path.Join(fourJoined, fourJoined), Path.Join(new string[] { path1, path2, path3, path4, path1, path2, path3, path4 }));
```

**Comment by jbhensley** [comment]
@dotnet-bot test this please
